### PR TITLE
Create UnitTest Environment + Deleting Example Assets + ListExtensions

### PR DESCRIPTION
**Pull request by mariokerndler**
- Deleted Example Assets.
- Created UnitTest Environemt. (Tested how it all works, with a few examples - PlayMode/EditMode)
- Created a library for ListExtensions (A few new methods that extends the generic list)

### CODE REVIEW

File: Assets/Tests/EditMode/ListExtensionTests.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+        [Test]
+        public void TestIsNullOrEmptz_NullList_True()
+        {
+            List<string> testList = null;
+            Assert.IsTrue(testList.IsNullOrEmpty());
+        }
+
+        [Test]
+        public void TestIsNullOrEmptz_FilledList_False()
```
Comment (Mocca101) [inline Assets/Tests/EditMode/ListExtensionTests.cs]: Emptz should prob. be Empty

File: Assets/Tests/EditMode/ListExtensionTests.cs [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+        [Test]
+        public void TestIsNullOrEmptz_EmptyList_True()
+        {
+            List<string> testList = new List<string>();
+            Assert.IsTrue(testList.IsNullOrEmpty());
+        }
+
+        [Test]
+        public void TestIsNullOrEmptz_NullList_True()
```
Comment (Mocca101) [inline Assets/Tests/EditMode/ListExtensionTests.cs]: Emptz should prob. be Empty

File: Assets/Tests/EditMode/ListExtensionTests.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            Assert.IsTrue(testList.IsNullOrEmpty());
+        }
+
+        [Test]
+        public void TestIsNullOrEmptz_FilledList_False()
+        {
+            List<int> testList = new List<int>(){ 1, 2, 3};
+            Assert.IsFalse(testList.IsNullOrEmpty());
+        }
+    }
```
Comment (Mocca101) [inline Assets/Tests/EditMode/ListExtensionTests.cs:102]: Maybe add a test to assert that an exception is thrown when the list is null?

Comment (mariokerndler) [reply Assets/Tests/EditMode/ListExtensionTests.cs:102]: If the list is null e.g.: List<int> testList;
the extension method IsNullOrEmpty() cant be called, because there is nothing assigned to the list.

**Review by Mocca101 — APPROVED** [review]
